### ISSUE TITLE
how to save image_data with bounding boxes as '.jpg'  in object_detection task

### ISSUE THREAD

**Issue by erinliu0428** [apple/turicreate#340]
In the object_detection task, we use
`data['image_with_ground_truth'] =  tc.object_detector.util.draw_bounding_boxes(data['image'], data['annotations'])`
to create a  turicreate.Image  type data which contains image  data with bounding boxes. So  how to save the turicreate.Image  type  data  as  '.jpg' or '.png' ?  The file '~/turicreate/data_structures/image.py '   doesn't  have a function like this

**Comment by erinliu0428** [comment]
I know already.We can use
`img = data['image_with_ground_truth']
 img._to_pil_image.save('test.jpg')`

**Comment by gustavla** [comment]
Hi @erinliu0428! This is definitely missing from `tc.Image` (along with other basic functionality that we hope to add). I think what you are describing works, but it is not part of the official API, so it could break at any new version without warning. A more API-stable approach would be 

```python
from PIL import Image
Image.fromarray(img.pixel_data).save('test.jpg')
```

I will leave this open and mark it as a feature request, since there should be an even simpler way.

**Comment by erinliu0428** [comment]
@gustavla Thank you for your attention !
And I have another basic need.After a object_detection model created, I want to predict some new pics. I need the new pics are parted from a video directly without be saved in a path.So if there is a way we can load video directly, or after loading video and parting images  by other toolkits  convert this type of image to tc.Image?   By the way,I'm curious about how tc.Image._image_data is generated. I thought if two pics are same size ,the shape of ._image_data  should be the same .But it turns out wrong.

**Comment by srikris** [comment]
That's actually a great feature request. I filed it at #342 and #343 to track this.

**Comment by shantanuchhabra** [comment]
Fixed by #709

**Comment by baizhenmao95** [comment]
@znation Thank you very much for your reply in #88 for "save the images at full size"; but when I tried the code:
`from PIL import Image
Image.fromarray(img.pixel_data).save('test.jpg')`
There was an AttributeError: 'SArray' object has no attribute 'pixel_data'
I also tried the code:
`img = data['image_with_ground_truth']
img._to_pil_image.save('test.jpg')`
but there was also the AttributeError: 'SArray' object has no attribute '_to_pil_image'

Is there any other things that I should do in advance?
Thank you!

**Comment by znation** [comment]
Hi @baizhenmao95, in the code examples there, `img` refers to a single [Image](https://apple.github.io/turicreate/docs/api/generated/turicreate.Image.html), whereas it appears in your code, `img` refers to an [SArray](https://apple.github.io/turicreate/docs/api/generated/turicreate.SArray.html) containing images. To operate on each one, you could pull them out individually, like:

```python
from PIL import Image
# note the [0] - this extracts just the first image from the SArray
Image.fromarray(img[0].pixel_data).save('test.jpg')
```

Or, you could run a lambda function over the SFrame to save all images in parallel. For this, you would need another SArray containing the desired filename (let's assume it's called "filename"):

```python
from PIL import Image
# make a temporary SFrame to associate images with filenames
temp_sf = tc.SFrame()
temp_sf["img"] = img
temp_sf["filename"] = filename
temp_sf.apply(lambda row: Image.fromarray(row["img"].pixel_data).save(row["filename"]))
```

**Comment by baizhenmao95** [comment]
Wow, that's cool! Thank you very much! I'll try it.

**Comment by PassantReyad** [comment]
Anybody can help me find out how to save a video file with bounding boxes as '.npy' after the object_detection task?
